hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ContainerReplicaOp.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.scm.container.replication;
+
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+
+/**
+ * Class to wrap details used to track pending replications.
+ */
+public class ContainerReplicaOp {
+
+  /**
+   * Enum representing different types of pending Ops.
+   */
+  public enum PendingOpType {
+    ADD, DELETE
```

Comment:
NIT: can we also add a type "ECRECONSTRUCT" , since EC reconstruction may also be a sort of option

Reply:
I am not sure we need ECReconstruct here. The purpose of an ECReconstruction is to create a new copy of some replicas. Its really a command like ReplicateBlock, only the command can create several new replicas at once.

For an ECReconstruction to recover indexes 2 and 3, we would just add 2 actions for ADD with the relevant index, and then the command would get trigger on the DN.

Reply:
after a deep thought , i did not find any problem about your approach. yes , we can treat ECReconstruct as a list of separate add options of a certain container.